### PR TITLE
Outbound Socket Support

### PR DESCRIPTION
**Pull request by italorossi**

### CODE REVIEW

**Comment by victor-torres** [comment]
Are you finishing this feature? There are some conflicts.

**Comment by italorossi** [comment]
Yes. It'll be available soon.

**Comment by italorossi** [comment]
@evangilo We may need to control when we want the calls to be hangup or wait to call finish when stopping the server.

**Comment by italorossi** [comment]
Guess we're ready to merge this. Waiting for tests.

File: greenswitch/esl.py [hunk 1]
```diff
@@ -187,33 +204,310 @@ def send(self, data):
...
+            try:
+                self.server.bind((self.bind_address, port))
+                bound = True
+                self.bound_port = port
+                break
+            except socket.error:
+                logging.info('Failed to bind to port %s, trying next in range...' % port)
+                continue
+        if not self.bound_port:
+            loggind.error('Could not bind server, no ports available.')
```
Comment (waldyr) [inline greenswitch/esl.py]: Typo: `loggind` <=> `logging`

Comment (victor-torres) [reply greenswitch/esl.py]: @waldyr is alive! :D

Comment (italorossi) [reply greenswitch/esl.py]: thanks, fixed

**Comment by dedayoa** [comment]
Well done @all. Thanks for the work on the Outbound socket capability.

**Pull request merged**